### PR TITLE
Remove redundant code related to can connection

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Delete all code related to can connection
Delete all validators due to Validation Throw API schema

### CODE REVIEW

**Review by LitvinenkoIra — APPROVED** [review]

File: src/components/can_cooperation/src/can_module.cc [hunk 1]
```diff
@@ -130,94 +159,18 @@ ProcessResult CANModule::ProcessMessage(application_manager::MessagePtr msg) {
...
-                 json_keys::kMethod)) {
-    function_name =
-        value[json_keys::kError][json_keys::kData][json_keys::kMethod]
-            .asCString();
-    msg->set_message_type(application_manager::MessageType::kErrorResponse);
-  } else {
-    DCHECK(false);
-    return ProcessResult::FAILED;
-  }
+  std::string function_name = ExtractFunctionAndAddMetadata(value, *msg);
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan Const?

File: src/components/functional_module/src/plugin_manager.cc [hunk 2]
```diff
@@ -173,27 +199,10 @@ ProcessResult PluginManager::ProcessHMIMessage(
...
-    } else if (value.isMember("error") && value["error"].isMember("data") &&
-               value["error"]["data"].isMember("method") &&
-               value["error"]["data"]["method"].isString()) {
-      msg_method = value["error"]["data"]["method"].asCString();
-    } else {
-      LOG4CXX_WARN(logger_,
-                   "Message with HMI protocol version can not be handled by "
-                   "plugin manager, because required 'method' field was not "
-                   "found, or was containing an invalid string.");
+    std::string msg_method = ExtractMethodName(value);
```
Comment (AByzhynar) [inline src/components/functional_module/src/plugin_manager.cc]: @LuxoftAKutsan Const?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/src/can_module.cc [hunk 3]
```diff
@@ -105,21 +95,60 @@ functional_modules::PluginInfo CANModule::GetPluginInfo() const {
   return plugin_info_;
 }
 
-ProcessResult CANModule::ProcessMessage(application_manager::MessagePtr msg) {
-  DCHECK(msg);
+const std::string ExtractFunctionAndAddMetadata(
+    const Json::Value& value, application_manager::Message& out_msg) {
+  if (value.isMember(json_keys::kMethod)) {
+    std::string function_name = value.get(json_keys::kMethod, "").asCString();
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan const

File: src/components/can_cooperation/src/can_module.cc [hunk 4]
```diff
@@ -105,21 +95,60 @@ functional_modules::PluginInfo CANModule::GetPluginInfo() const {
...
+    } else {
+      out_msg.set_message_type(application_manager::MessageType::kNotification);
+    }
+    return function_name;
+  }
+
+  if (value.isMember(json_keys::kResult)) {
+    // value[json_keys::kResult].isMember(json_keys::kMethod)
+    const Json::Value& result = value.get(json_keys::kResult, Json::Value());
+    std::string function_name = result.get(json_keys::kMethod, "").asCString();
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan const

File: src/components/can_cooperation/src/can_module.cc [hunk 5]
```diff
@@ -105,21 +95,60 @@ functional_modules::PluginInfo CANModule::GetPluginInfo() const {
...
-    LOG4CXX_ERROR(logger_, "Null pointer message received.");
-    return ProcessResult::FAILED;
+    out_msg.set_message_type(application_manager::MessageType::kResponse);
+    return function_name;
   }
 
+  if (value.isMember(json_keys::kError)) {
+    const Json::Value& error = value.get(json_keys::kError, Json::Value());
+    const Json::Value& data = error.get(json_keys::kData, Json::Value());
+    std::string function_name = error.get(json_keys::kMethod, "").asCString();
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan const

**Review by AKalinich-Luxoft — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/src/can_module.cc (hunk 5 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/can_module.cc]: @AKalinich-Luxoft e6d44a9

File: src/components/can_cooperation/src/can_module.cc (hunk 4 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/can_module.cc]: @AKalinich-Luxoft e6d44a9

File: src/components/can_cooperation/src/can_module.cc (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/can_module.cc]: @AKalinich-Luxoft e6d44a9

File: src/components/functional_module/src/plugin_manager.cc (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/functional_module/src/plugin_manager.cc]: @AByzhynar  e6d44a9

File: src/components/can_cooperation/src/can_module.cc (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/can_module.cc]: @AByzhynar  e6d44a9

**Review by AKalinich-Luxoft — APPROVED** [review]

File: src/components/can_cooperation/src/can_module.cc [hunk 6]
```diff
@@ -226,91 +182,18 @@ functional_modules::ProcessResult CANModule::HandleMessage(
...
-      msg->set_protocol_version(application_manager::ProtocolVersion::kV3);
-      NotifyMobiles(msg);
-      break;
-    }
-
-    case application_manager::MessageType::kRequest: {
-      if (function_name == functional_modules::hmi_api::sdl_activate_app) {
-        msg->set_protocol_version(application_manager::ProtocolVersion::kHMI);
-        return ModuleHelper::ProcessSDLActivateApp(value, *this);
+      bool is_valid = service()->ValidateMessageBySchema(*msg);
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan Const?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Pull request merged**